compiler/rustc_lint/src/lib.rs

```diff
@@ -321,6 +321,7 @@ fn register_builtins(store: &mut LintStore, no_interleave_lints: bool) {
         INVALID_CODEBLOCK_ATTRIBUTES,
         MISSING_DOC_CODE_EXAMPLES,
         PRIVATE_DOC_TESTS,
+        MISSING_CRATE_LEVEL_DOCS,
```

Comment:
This should not be mixed in with the cleanup; it's the same sort of change as https://github.com/rust-lang/rust/pull/77364, which I'm still not convinced is the right change.

Reply:
It's only emitted by rustdoc, so the context is a bit different there. This whole PR wouldn't be possible without this change because otherwise I'll need to still handle this lint differently than the others. However, because it was a bit out of the scope of this PR, I have put this change in a commit on its own (it's the first commit, you can check ;) ).